proof/dleq_test.go

```diff
@@ -0,0 +1,88 @@
+package proof
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/edwards"
+	"github.com/dedis/crypto/random"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDLEQProof(t *testing.T) {
+	suite := edwards.NewAES128SHA256Ed25519(false)
+	n := 10
+	var good []int
+	var bad []int
+
+	for i := 0; i < n; i++ {
+		// Create some random secrets and base points
+		x := suite.Scalar().Pick(random.Stream)
+		g, _ := suite.Point().Pick([]byte(fmt.Sprintf("G%d", i)), random.Stream)
+		h, _ := suite.Point().Pick([]byte(fmt.Sprintf("H%d", i)), random.Stream)
+
+		proof, xG, xH, err := NewDLEQProof(suite, g, h, x)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if err := proof.Verify(suite, g, h, xG, xH); err == nil {
+			good = append(good, i)
+		} else {
+			bad = append(bad, i)
+		}
+	}
+	require.Equal(t, len(bad), 0)
```

Comment:
if you don't need `good` and `bad` for anything else, I'd just `require.Nil(proof.Verify` above and don't use `good` and `bad` at all. Those are very old concepts anyway which are very relative ;)